### ISSUE TITLE
Two- and Four-way coupled

### ISSUE THREAD

**Issue by gforney** [firemodels/fds#1217]
```
Application Version: 5.1.1
SVN Revision Number: 6385
Compile Date:        22 jun 2010
Operating System:    normal PC


Dear K. McGrattan et all, 

After my last post about the pressure drop over a cloud of droplets I’ve continued
to investigate the FDS sprinkler droplet simulation procedures. 

In several articles and books I noticed that there are two procedures used in particle
simulations of which I’m unsure whether they are applied in the FDS algorithm.  These
procedures are indicated as ‘two way coupled’ and ‘four way coupled’ Euler Lagrangian
Simulations. 
The two way coupled procedure uses coupling iterations to recalculate the fluid flow
of the gas with the effects of the particles and then recalculates the particle movements
until convergence is achieved. I’ve found the Fb force from equation (9.1) used in
equation (3.3). But I’m unsure if the new flow field is then used to recalculate the
particle trajectories in an iterative manner. By looking at the code file main.f90
I am unable to locate such an iterative convergence procedure.
The four way coupling procedure described in the literature concerns taking into account
particle collisions. Especially the article by Vreman et all, which is attached, concludes
this can be an important effect in flow fields containing particles.

In the links below I have added two articles which regard two and four way coupling
as an important part of particle simulations. The two attachments are  scans from the
book ‘Modern drying technology, Tsotsas&Mujumdar, 2007’. 
Although I haven’t been able to find a precise description of the two techniques these
articles are in my opinion strong indications that they are important to take into
consideration.

In conclusion I would like to ask whether FDS uses these two forms of simulation techniques.
Or perhaps they are on schedule to be incorporated in the 6th version?
Best regards,

Romboud van de Woestijne
Master student of physics
Delft university of technology, Holland

Two- and four-way coupled Euler-Lagrangian Large-Eddy Simulation of turbulent particle-laden
channel flow
http://doc.utwente.nl/59747/

two-way coupled large-eddy simulations of the gas-solid flow in cyclone separators
http://onlinelibrary.wiley.com/doi/10.1002/aic.11418/abstract


```

Original issue reported on code.google.com by `romboudvdw` on 2010-11-02 14:47:06

**Comment by gforney** [comment]
```
Do not post copyrighted material.  Should a publisher file a DMCA complaint, FDS would
could lose its access to GoogleCode .
```

Original issue reported on code.google.com by `drjfloyd` on 2010-11-02 15:11:08

**Comment by gforney** [comment]
```
We are in the process of changing our droplet/particle momentum transfer scheme for
FDS 6. We will take a look at the articles. Thanks.
```

Original issue reported on code.google.com by `mcgratta` on 2010-11-02 21:29:26

**Comment by gforney** [comment]
```
Romboud,

Thanks for pointing us to Vreman's article.  I will take a look when I get time.  Here
are some notes on our current implementation.  It does not account for collisions,
but the two-way fluid coupling is handled through a direction solution of the fluid-particle
ODE without the need for iteration.

https://sites.google.com/site/randymcdermott/drag_solution.pdf

Best,
Randy
```

Original issue reported on code.google.com by `randy.mcdermott` on 2010-11-02 21:35:48

**Comment by gforney** [comment]
```
By the way, should this not be posted to the discussion forum?  I am marking this as
won't fix, because I do not see a bug report or enhancement request.

This does not mean we will not keep developing the particle method.  Just that we are
cleaning things up for FDS6 right now.
```

Original issue reported on code.google.com by `randy.mcdermott` on 2010-11-02 21:53:22

**Comment by gforney** [comment]
```
Dear all,

Firstly I'm sorry for posting copyrighted material. I will be more carefull in the
future.

Very nice to read that you will have a look into the four-way coupled method. I am
as current unable to provide a detailed technical explaination of it. But I will have
a further investigation and hope I can provide you guys with a good book or article.

Randy, thank you for your article about the two way coupled method implemented in FDS.

Best regards,

Romboud
```

Original issue reported on code.google.com by `romboudvdw` on 2010-11-04 08:55:05

**Comment by gforney** [comment]
```
No problem -- I make this mistake often. Google is very sensitive to it because they
are being sued by just about everyone these days. I wish people would go back to bashing
Microsoft.

As for droplets, Randy's notes present a relatively simple method to couple the droplet
and air movement. Currently in FDS, we move the droplets, then we "move" the air using
the initial estimate of the droplet drag. This is OK for relatively big droplets, but
it becomes a problem when the droplets are small and moving fast, like those from a
mist nozzle.
```

Original issue reported on code.google.com by `mcgratta` on 2010-11-04 12:13:29

**Comment by gforney** [comment]
```
Dear Kevin,

After this discussion I have done some further investigation on two-way and four way-coupled
particle CFD simulations. Unfortunately I have not yet found a proper description on
how to specifically program these effects. 
But I was wondering, in your first reaction on this post you mentioned that you are
changing the droplet/particle momentum transfer scheme for FDS 6. Were you already
in the process of implementing a two way coupled, possibly four-way, algorithm? 
I have asked my professor about it and he is fairly convinced the interaction on the
particles with the air will be a more important aspect of the interaction with a smoke
layer then the energy and mass transfer between the droplets and the smoke layer. For
the situation of the smoke layer behaviour under influence of a sprinkler.

I've done some further reeding and indeed the droplet sizes with respect to the turbulence(kolmogorov
scales) within the fluid seem important. At which minimal droplet diameter approximately
do you estimate that FDS currently returns a good prediction?

Best regards,

Romboud
```

Original issue reported on code.google.com by `romboudvdw` on 2010-11-16 15:51:40

**Comment by gforney** [comment]
```
The algorithm (link in Comment 3) has been implemented for FDS 6. We are currently testing
it. I think it is what you call 2 way coupling. I hesitate to say for what range of
droplet sizes FDS produces a "good" solution. I can say that we have mainly used the
droplet algorithm for sprinklers whose droplet sizes are hundreds of microns to a few
mm. 
```

Original issue reported on code.google.com by `mcgratta` on 2010-11-16 15:56:54
